aiida_mlip/calculations/train.py

```diff
@@ -0,0 +1,170 @@
+        Check if the inputs are valid.
+    prepare_for_submission(folder: Folder) -> CalcInfo:
+        Create the input files for the `CalcJob`.
+    """
+
+    DEFAULT_OUTPUT_FILE = "aiida-stdout.txt"
+    DEFAULT_INPUT_FILE = "aiida.xyz"
+    LOG_FILE = "aiida.log"
+
+    @classmethod
+    def define(cls, spec: CalcJobProcessSpec) -> None:
+        """
+        Define the process specification, its inputs, outputs and exit codes.
+
+        Parameters
+        ----------
+        spec : `aiida.engine.CalcJobProcessSpec`
+            The calculation job process spec to define.
+        """
+        super().define(spec)
+        spec.inputs.validator = validate_inputs
+        # Define inputs
+        spec.input(
+            "mlip_config",
+            valid_type=JanusConfigfile,
+            required=False,
+            help="Mlip architecture to use for calculation, defaults to mace",
+        )
+        spec.output("model", valid_type=ModelData)
+        spec.output("compiled_model", valid_type=SinglefileData)
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        spec.output("logs", valid_type=FolderData)
+        spec.output("checkpoints", valid_type=FolderData)
+        spec.default_output_node = "results_dict"
+        # Exit codes
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        cmd_line = {}
+
+        cmd_line["mlip-config"] = "mlip_train.yaml"
+        config_parse = self.inputs.config.get_content()
+        mlip_dict = self.inputs.mlip_config.as_dictionary()
+        # Copy config file content inside the folder where the calculation is run
+        with folder.open("mlip_config.yaml", "w", encoding="utf-8") as configfile:
+            configfile.write(config_parse)
+
+        model_dir = Path(mlip_dict.get("model_dir", "."))
+        model_output = model_dir / f"{mlip_dict['name']}.model"
+        compiled_model_output = model_dir / f"{mlip_dict['name']}_compiled.model"
+
+        codeinfo = datastructures.CodeInfo()
+
+        # Initialize cmdline_params with train command
+        codeinfo.cmdline_params = ["train"]
+
+        for flag, value in cmd_line.items():
+            codeinfo.cmdline_params += [f"--{flag}", str(value)]
```

Comment:
Given this is such a common pattern to all of these would it be best to turn this into a function?